tests/unit/tests_standard_lib/test_index_tests/test_test_generator.py

```diff
@@ -0,0 +1,506 @@
+def raise_warning_mock(monkeypatch):
+    rw = MagicMock()
+    monkeypatch.setattr(
+        "pytest_splunk_addon.standard_lib.index_tests.test_generator.raise_warning",
+        rw,
+    )
+    return rw
+
+
+def test_generate_tests_without_conf_file(sample_generator_mock, caplog):
+    sample_generator_mock.return_value = sample_generator_mock
+    sample_generator_mock.get_samples.return_value = {"tokenized_events": []}
+    list(
+        IndexTimeTestGenerator().generate_tests(
+            True, "fake_app_path", "fake_config_path", "key_fields"
+        )
+    )
+    assert caplog.messages == [
+        "Index Time tests cannot be executed using eventgen.conf, pytest-splunk-addon-data.conf is required."
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_type, tokenized_events, expected_output",
+    [
+        (
+            "key_fields",
+            [
+                sample_event(
+                    metadata={
+                        "identifier": "sample.1",
+                        "host": ["localhost", "remotehost"],
+                        "host_prefix": "p_",
+                    },
+                    key_fields={"host": "localhost"},
+                )
+            ],
+            [
+                (
+                    sample_event(
+                        metadata={
+                            "identifier": "sample.1",
+                            "host": ["localhost", "remotehost"],
+                            "host_prefix": "p_",
+                        },
+                        key_fields={"host": ["p_localhost"]},
+                    ),
+                    "sample.1",
+                    ["localhost", "remotehost"],
+                )
+            ],
+        ),
+        (
+            "_time",
+            [
+                sample_event(
+                    metadata={
+                        "identifier": "sample.2",
+                        "host": "localhost",
+                        "timestamp_type": "event",
+                    },
+                )
+            ],
+            [
+                (
+                    sample_event(
+                        metadata={
+                            "identifier": "sample.2",
+                            "host": "localhost",
+                            "timestamp_type": "event",
+                        },
+                    ),
+                    "sample.2",
+                    ["localhost"],
+                )
+            ],
+        ),
+    ],
+)
+def test_generate_tests_triggers_generate_params(
+    sample_generator_mock,
+    sample_event_mock,
+    test_type,
+    tokenized_events,
+    expected_output,
+):
+    sample_generator_mock.return_value = sample_generator_mock
+    sample_event_mock.copy.side_effect = lambda x: deepcopy(x)
+    sample_generator_mock.get_samples.return_value = {
+        "tokenized_events": tokenized_events,
+        "conf_name": "psa-data-gen",
+    }
+    with patch.object(
+        IndexTimeTestGenerator,
```

Comment:
There are a lot of similar patches. You can extract function `funciton_name`  and `lambda` as parameters